frontend/packages/ux-editor/src/utils/formLayoutsUtils.ts

```diff
@@ -52,17 +52,18 @@ export const addOrRemoveNavigationButtons = async (
         await callback(name, layoutWithNavigation);
       }
     }
-    if (currentLayoutName) {
-      // Add navigation buttons to the current layout if they are not present, and run callback
-      let currentLayout = layouts[currentLayoutName];
-      if (!hasNavigationButtons(currentLayout) && currentLayoutName !== receiptLayoutName){
-        const navButtonsId = generateComponentId(ComponentType.NavigationButtons, layouts);
-        currentLayout = addNavigationButtons(currentLayout, navButtonsId);
-        updatedLayouts[currentLayoutName] = currentLayout;
-      }
-      await callback(currentLayoutName, currentLayout);
+  }
+  if (currentLayoutName) {
+    // Add navigation buttons to the current layout if they are not present, and run callback
+    let currentLayout = layouts[currentLayoutName];
+    if (!hasNavigationButtons(currentLayout) && currentLayoutName !== receiptLayoutName){
+      const navButtonsId = generateComponentId(ComponentType.NavigationButtons, layouts);
+      currentLayout = addNavigationButtons(currentLayout, navButtonsId);
+      updatedLayouts[currentLayoutName] = currentLayout;
     }
+    await callback(currentLayoutName, currentLayout);
   }
+
```

Comment:
Denne funksjonen begynner å bli ganske uoversiktlig. Jeg skrev om den ene testen til å ha `currentLayoutName` når det er kun én layout i tillegg til kvittering, og da feiler den fordi callback blir kalt to ganger. Det er mulig dette ikke er en situasjon som oppstår i koden slik den er nå (og at det derfor ikke feiler når man tester manuelt), men vi kan ikke forutsette at fremtidige utviklere er klar over det. Jeg foreslår derfor at vi skriver om funksjonen litt, omtrent slik som dette:
```
export const addOrRemoveNavigationButtons = async (
  layouts: IFormLayouts,
  callback: (layoutName: string, layout: IInternalLayout) => Promise<void>,
  currentLayoutName?: string,
  receiptLayoutName?: string
): Promise<IFormLayouts> => {
  if (currentLayoutName && !layouts[currentLayoutName]) {
    throw new Error(`Layout with name ${currentLayoutName} does not exist.`);
  }

  const allLayouts = deepCopy(layouts);
  let layoutsToUpdate: string[] = [];

  // Update layouts to have navigation buttons if there are multiple layouts, or remove them if there is the only one.
  const allLayoutNames = Object.keys(layouts);
  const layoutsThatShouldHaveNavigationButtons = allLayoutNames.filter((name) => name !== receiptLayoutName);
  if (layoutsThatShouldHaveNavigationButtons.length === 1) {
    // There is only one layout
    const name = layoutsThatShouldHaveNavigationButtons[0];
    const layout = removeComponentsByType(layouts[name], ComponentType.NavigationButtons);
    layoutsToUpdate.push(name);
    allLayouts[name] = layout;
  } else {
    // There are multiple layouts
    for (const name of layoutsThatShouldHaveNavigationButtons) {
      const layout = layouts[name];
      if (!hasNavigationButtons(layout)) {
        const navButtonsId = generateComponentId(ComponentType.NavigationButtons, layouts);
        allLayouts[name] = addNavigationButtons(layout, navButtonsId);
        layoutsToUpdate.push(name);
      }
    }
  }
  currentLayoutName && layoutsToUpdate.push(currentLayoutName); // Always update the current layout if it is set
  layoutsToUpdate = removeDuplicates(layoutsToUpdate); // Remove duplicates so that callback is only called once for each layout
  await Promise.all(layoutsToUpdate.map((name) => callback(name, allLayouts[name])));
  return allLayouts;
};
```